v1/lib/core/nav/SideNav.js

```diff
@@ -144,19 +144,51 @@ class SideNav extends React.Component {
                 }
               }, false);
 
-              function createToggler(togglerSelector, targetSelector, className) {
+              function createModalToggler(togglerSelector, className) {
                 var toggler = document.querySelector(togglerSelector);
-                var target = document.querySelector(targetSelector);
 
                 if (!toggler) {
                   return;
                 }
 
-                toggler.onclick = function(event) {
+                toggler.onclick = isIOS() ? getIOSToggleHandler(className) : getToggleHandler(className);
+              }
+
+              function isIOS() {
+                var iOSDevices = ['iPad Simulator', 'iPhone Simulator', 'iPod Simulator', 'iPad', 'iPhone', 'iPod'];
```

Comment:
I can only comment from code perspective, but this part can definitely be refactored with something like

```js
iosDevices.indexOf(navigator.platform)
```

instead of the whole pop and while loop.

This also doesn't completely fix #877. #877 is related to glitchy scrolling on Mac Safari too. I don't have a Mac to test it out, unfortunately






cc @yangshun @JoelMarcey